test/e2e/fishapp/dynamic_stack.go

```diff
@@ -655,6 +680,56 @@ func (s *DynamicStack) createResourcesForServices(ctx context.Context, f *framew
 			}
 			Expect(len(waitErrors)).To(BeZero())
 		})
+
+		By("check all VirtualRouters in AWS", func() {
+			var checkErrors []error
+			checkErrorsMutex := &sync.Mutex{}
+			var wg sync.WaitGroup
+			for i := 0; i != s.VirtualServicesCount; i++ {
+				wg.Add(1)
+				go func(serviceIndex int) {
+					defer wg.Done()
+					vr := s.createdServiceVRs[serviceIndex]
+					err := f.VRManager.CheckVirtualRouterInAWS(ctx, s.mesh, vr)
+					if err != nil {
+						checkErrorsMutex.Lock()
+						checkErrors = append(checkErrors, errors.Wrapf(err, "VirtualRouter: %v", k8s.NamespacedName(vr).String()))
+						checkErrorsMutex.Unlock()
+						return
+					}
+				}(i)
+			}
+			wg.Wait()
+			for _, checkErr := range checkErrors {
+				f.Logger.Error("failed to check all VirtualRouters in AWS", zap.Error(checkErr))
+			}
+			Expect(len(checkErrors)).To(BeZero())
+		})
+
+		By("check all VirtualService in AWS", func() {
+			var checkErrors []error
+			checkErrorsMutex := &sync.Mutex{}
+			var wg sync.WaitGroup
+			for i := 0; i != s.VirtualServicesCount; i++ {
+				wg.Add(1)
+				go func(serviceIndex int) {
+					defer wg.Done()
+					vs := s.createdServiceVSs[serviceIndex]
+					err := f.VSManager.CheckVirtualServiceInAWS(ctx, s.mesh, vs)
+					if err != nil {
+						checkErrorsMutex.Lock()
+						checkErrors = append(checkErrors, errors.Wrapf(err, "VirtualService: %v", k8s.NamespacedName(vs).String()))
+						checkErrorsMutex.Unlock()
+						return
```

Comment:
no re-try / wait incase the convergence time is longer?

Reply:
we don't need retry here, since we waited the virtualService to be active. :D 
in the future, we need to add a "observedGeneration" in CRD's status, if we want to test for change.

Reply:
Ah, that saves this indirectly :D - would be good to add later for sure to not rely on a different subsystem for resources readiness here. Looks good!